pkg/api/datasources.go

```diff
@@ -17,11 +17,27 @@ func GetDataSources(c *m.ReqContext) Response {
 		return Error(500, "Failed to query datasources", err)
 	}
 
+	dsFilterQuery := m.DatasourcesPermissionFilterQuery{
+		User:        c.SignedInUser,
+		Datasources: query.Result,
+	}
+
+	var datasources []*m.DataSource
+	if err := bus.Dispatch(&dsFilterQuery); err != nil {
+		if err != bus.ErrHandlerNotFound {
+			return Error(500, "Could not get datasources", err)
+		}
+
+		datasources = query.Result
+	} else {
+		datasources = dsFilterQuery.Result
+	}
```

Comment:
I don't think we need to filter here in the /api/datasources as this api is protected by org admin permissions. We do need this  in a next phase, but then I think we should break be above lines out and reuse it in getFrontendSettings